### DISCUSSION TITLE
How to disable `console.log` during my tests?

### DISCUSSION THREAD

**Discussion by keinsell** [avajs/ava#3045] [Q&A]
I'm using `signale` as logger in my project and I wish to suppress all console.logs during tests, I'm not able to find any information how to do that - did you have done such thing before?

**Comment by novemberborn** [comment]
AVA doesn't call `console.log`, are you asking because there's other code in your project that is logging? Perhaps that can be disabled  separately?

I suppose you could try and override the methods on the global `console` so that they don't actually log anything.

**Reply by keinsell** [reply]
Yeah, generally I'm using `signale` as logger in my project and that causing some mess. However, I've figured it out like 2-3 days ago with usage of `sinon`, all that I needed to do was `sinon.stub(console)`. Problem with `signale` is still persistent but I think that's issue with package rather than `ava`. My question here was generic how to do such thing under `ava`.

**Reply by novemberborn** [reply]
There is no option to silence stdout/stderr from the test workers. AVA does route the output, so we _could_ add it. I'm not sure it's worthwhile though. It may add confusion instead if you forget you silenced the output and are trying to debug something.

I do use AVA to run integration tests against services, but I actually run the service in a child process and have the test harness intercept its logs.
